packages/react-app/src/components/quest.tsx

```diff
@@ -97,6 +98,22 @@ export default function Quest({
   const [claimDeposit, setClaimDeposit] = useState<TokenAmountModel | undefined>();
   const [challengeDeposit, setChallengeDeposit] = useState<TokenAmountModel | null>();
   const { below } = useViewport();
+  const { transaction } = useTransactionContext();
+
+  useEffect(() => {
```

Comment:
Here is what the useEffect should look like to have all transactions update UI after completion: 
`
  useEffect(() => {
      if (transaction?.status === ENUM_TRANSACTION_STATUS.Confirmed) {
        // If tx completion impact Quest bounty, update it
        if (
          transaction?.transactionType === 'ClaimChallengeResolve' ||
          transaction?.transactionType === 'ClaimExecute' ||
          transaction?.transactionType === 'QuestFund' ||
          transaction?.transactionType === 'QuestReclaimFunds'
        ) {
          setBounty(null);
          setTimeout(() => {
            if (questData.address && questData.rewardToken) {
              fetchBalanceOfQuest(questData.address, questData.rewardToken);
            }
          }, 500);
        }
  
        // If tx completion impact Claims, update them
        if (
          transaction?.transactionType === 'ClaimChallengeResolve' ||
          transaction?.transactionType === 'ClaimExecute' ||
          transaction?.transactionType === 'ClaimSchedule' ||
          transaction?.transactionType === 'ClaimChallenge'
        ) {
          const oldClaimUpdated = claimUpdated;
          setTimeout(() => {
            setClaimUpdate(oldClaimUpdated + 1);
          }, 500);
        }
      }
    }, [transaction?.transactionType, transaction?.status]);
`